blocks/handler_test.go

```diff
@@ -51,11 +54,14 @@ func createBlockData(t *testing.T, layerID types.LayerID, txIDs []types.Transact
 
 func Test_HandleBlockData_MalformedData(t *testing.T) {
 	th := createTestHandler(t)
-	layerID := types.NewLayerID(99)
-	txIDs := createTransactions(t, max(10, rand.Intn(100)))
+	assert.ErrorIs(t, th.HandleSyncedBlock(context.TODO(), []byte("malformed")), errMalformedData)
+}
 
-	_, data := createBlockData(t, layerID, txIDs)
-	assert.ErrorIs(t, th.HandleSyncedBlock(context.TODO(), data[1:]), errMalformedData)
+func Test_HandleBlockData_InvalidRewards(t *testing.T) {
+	th := createTestHandler(t)
+	buf, err := codec.Encode(&types.Block{})
+	require.NoError(t, err)
+	require.Error(t, th.HandleSyncedBlock(context.TODO(), buf))
```

Comment:
i think we should best-effort check for exact error here if it's not too much trouble.
defining an new error for invalid rewards?